### PR TITLE
Middlewares

### PR DESCRIPTION
**Pull request by ranuccimariano**
## Summary

Moved user creation validations to a middleware

## Trello Card

https://trello.com/c/vcasC58r

### CODE REVIEW

File: app/middlewares/user_validation.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const { checkSchema, validationResult } = require('express-validator');
+
+const error = require('../errors');
+
+const { userCreationError } = error;
+
+exports.userValidationRules = () => [
+  checkSchema({
+    first_name: {
+      notEmpty: true,
```
Comment (molinajulian) [inline app/middlewares/user_validation.js]: add `isString:true`, `required:true`

File: app/middlewares/user_validation.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      errorMessage: "firstName can't be blank"
+    },
+    last_name: {
+      notEmpty: true,
+      errorMessage: "lastName can't be blank"
+    },
+    email: {
+      notEmpty: true,
+      matches: {
```
Comment (molinajulian) [inline app/middlewares/user_validation.js]: I think that exist `isEmail` but Im not sure :thinking:

File: app/middlewares/user_validation.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        errorMessage: 'you may only use email addresses from wolox domain'
+      }
+    },
+    password: {
+      isLength: {
+        errorMessage: 'Password should be at least 8 characters long',
+        options: { min: 8 }
+      },
+      isAlphanumeric: {
+        errorMessage: 'Password should be alphanumeric'
```
Comment (molinajulian) [inline app/middlewares/user_validation.js]: Maybe you can create a file for errors messages? And create a folder `/schemas` with your schemas. What do you think?

File: app/routes.js [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
 const { getAlbums, getAlbumPhotos } = require('./controllers/albums');
 const { postUser } = require('./controllers/users');
+const { userValidationRules, validateUser } = require('./middlewares/user_validation');
 
 exports.init = app => {
   app.get('/health', healthCheck);
   app.get('/albums', getAlbums);
   app.get('/albums/:id/photos', getAlbumPhotos);
-  app.post('/users', postUser);
+  app.post('/users', [userValidationRules(), validateUser], postUser);
```
Comment (molinajulian) [inline app/routes.js]: The `()` is not necessary

File: app/schemas/users.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+exports.userSignUpSchema = {
+  first_name: {
+    in: ['body'],
+    exists: true,
+    isString: true,
+    notEmpty: true,
+    trim: true,
+    errorMessage: "The key 'first_name' it's required, must be of type 'string' and can't be blank"
```
Comment (molinajulian) [inline app/schemas/users.js]: Would create a file for errorMessages? Maybe `constants.js` or similar

File: app/controllers/users.js [hunk 6]
```diff
@@ -9,34 +7,15 @@ const { hashPassword } = require('../services/users');
...
-              return user;
-            })
-            .catch(err => {
-              logger.error(util.inspect(err));
-              throw userCreationError(err.message);
-            })
-        )
+      User.create({ firstName, lastName, email, password })
+        .then(user => {
+          logger.info(`Usuario creado para: ${firstName} ${lastName}`);
```
[the remark this answers is not in the document]
Comment (ranuccimariano) [reply app/controllers/users.js]: Sorry, i thought i've already changed that :sweat_smile:

**Pull request merged**